docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+
+-   android
+-   chrome
+-   edge
+-   firefox
+-   ie11
+-   ios
+-   node
+-   node8
+-   safari
+
+## Provided Features
+
+These features are provided by the build system to help identify a specific environment or mode of operation.
+
+| Feature Flag        | Description                                                                                                                                                                                                                                              |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `debug`             | Provides a way to create a code path for code that is only usable when debugging or providing enhanced diagnostics that are not desired in a _production_ build. Defaults to `true` but should be configured statically as `false` in production builds. |
+| `host-browser`      | Determines if the current environment contains a `window` and `document` object in the global context, therefore it is generally safe to assume the code is running in a browser environment.                                                            |
+| `host-node`         | Attempts to detect if the environment appears to be a node environment.                                                                                                                                                                                  |
+| `build-time-render` | Statically defined by the build-time rendering system during build-time rendering.                                                                                                                                                                       |
+
+# Build-time Rendering
+
+BTR renders a route to HTML during the build process and in-lines critical CSS and assets needed to display the initial view. This allows Dojo to pre-render the initial HTML used by a route and inject it directly into the page immediately. Skipping the initial render by creating it during the build process gives us many of the same benefits of server side rendering (SSR) such as performance gains and search engine optimization without the complexities of SSR.
+
+## Using BTR
+
+First make sure `index.html` includes a DOM node with an `id` attribute. This node will be used by Dojo's virtual DOM to compare and render the application's HTML. BTR requires this setup so it can render the HTML generated at build time. This creates a very fast and responsive initial rendering of the route. Once the route has had a change to load and render the virtual DOM is able to diff against the BTR.
```

Comment:
"Once the route has had a change to load and render the virtual DOM is able to diff against the BTR" I'm not sure exactly what you mean by this?

Reply:
I *think* I was trying to say that BTR is built against routes. Definitely needs better wording.